app/models/concerns/profile_tailoring.rb

```diff
@@ -33,4 +33,19 @@ def update_os_minor_version(version)
 
     update!(os_minor_version: version)
   end
+
+  def tailoring_valid?
+    rule_groups_for_rules = RuleGroup.where(id: rules.includes(:rule_group)
+                                     .references(:rule_group).flat_map(&:rule_group_ancestor_ids))
```

Comment:
I guess it's safe to shadow here the `rule_groups` variable as the method is short enough.